src/lib/controller/message_handler.ts

```diff
@@ -57,6 +57,48 @@ export class ControllerMessageHandler {
         const success = await driver.controller.isFailedNode(message.nodeId);
         return { success };
       }
+      case ControllerCommand.getAssociationGroups: {
+        const groups = {};
+        await driver.controller
+          .getAssociationGroups(message.nodeId)
+          .forEach((value, key) => (groups[key] = value));
+        return { groups };
+      }
+      case ControllerCommand.getAssociations: {
+        const associations = {};
+        await driver.controller
+          .getAssociations(message.nodeId)
+          .forEach((value, key) => (associations[key] = value));
+        return { associations };
+      }
+      case ControllerCommand.isAssociationAllowed: {
+        const success = await driver.controller.isAssociationAllowed(
+          message.nodeId,
+          message.group,
+          message.association
+        );
+        return { success };
```

Comment:
Should this be allowed?
```suggestion
        const allowed = await driver.controller.isAssociationAllowed(
          message.nodeId,
          message.group,
          message.association
        );
        return { allowed };
```